controllers/appwrapper_controller.go

```diff
@@ -120,17 +120,24 @@ func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	demandPerInstanceType := r.discoverInstanceTypes(ctx, &appwrapper)
-	if ocmSecretRef := r.Config.OCMSecretRef; ocmSecretRef != nil {
-		return r.scaleMachinePool(ctx, &appwrapper, demandPerInstanceType)
-	} else {
-		switch strings.ToLower(r.Config.MachineSetsStrategy) {
-		case "reuse":
-			return r.reconcileReuseMachineSet(ctx, &appwrapper, demandPerInstanceType)
-		case "duplicate":
-			return r.reconcileCreateMachineSet(ctx, &appwrapper, demandPerInstanceType)
+	status := appwrapper.Status.State
+	allconditions := appwrapper.Status.Conditions
+	for _, condition := range allconditions {
+		if status == "Pending" && strings.Contains(condition.Message, "Insufficient") {
```

Comment:
Thanks! I would create a func called:  `func containsInsufficientCondition(allconditions)` that loops through all conditions of an AW. this simplifies the code and helps us check more or less conditions in the future.